siliconcompiler/report/Home.py

```diff
@@ -0,0 +1,601 @@
+        # widget does not update during the streamlit.experimental_rerun.
+        # We need to keep track of the 'true' flip of the toggle to make sure
+        # everything is synced
+        if 'right after rerun' in streamlit.session_state and \
+           streamlit.session_state['right after rerun']:
+            transpose = streamlit.session_state['transpose']
+            streamlit.session_state['right after rerun'] = False
+        streamlit.session_state['transpose'] = transpose
+
+
+def show_flowgraph():
+    """
+    Displays the header and toggle for the flowgraph, and the flowgraph itself.
+    This function shows the flowgraph. If the toggle is flipped, the flowgraph
+    will disappear.
+    """
+    home_tab_cols = streamlit.columns([0.4, 0.6], gap="large")
+    flowgraph_col, metrics_and_nodes_info_col = home_tab_cols
+
+    with flowgraph_col:
+        headerCol, toggleCol = streamlit.columns([0.7, 0.3], gap="large")
+        with headerCol:
+            streamlit.header('Flowgraph')
+
+        with toggleCol:
+            streamlit.markdown("\n")
+            fg_toggle = st_toggle_switch(label=" ",
+                                         key="flowgraph_toggle",
+                                         default_value=True,
+                                         label_after=True,
+                                         # the colors are optional
+                                         inactive_color=inactive_toggle_color,
+                                         active_color=active_toggle_color,
+                                         track_color=track_toggle_color)
+            streamlit.session_state['flowgraph'] = fg_toggle
+
+            if not streamlit.session_state['flowgraph']:
+                streamlit.session_state['right after rerun'] = True
+                streamlit.experimental_rerun()
+
+        # need to update dynamically, could use number of attributes displayed
+        # + offset
+        config = Config(width='100%',
+                        directed=True,
+                        physics=False,
+                        hierarchical=True,
+                        clickToUse=True,
+                        nodeSpacing=150,
+                        levelSeparation=100,
+                        sortMethod='directed')
+
+        node_from_flowgraph = agraph(nodes=nodes,
+                                     edges=edges,
+                                     config=config)
+    return node_from_flowgraph, metrics_and_nodes_info_col
+
+
+def dont_show_flowgraph():
+    """
+    Displays the header and toggle for the flowgraph, and the flowgraph itself.
+    This function doesn't show the flowgraph. If the toggle is flipped, the
+    flowgraph will re-appear.
+    """
+    home_tab_cols = streamlit.columns([0.1, 0.9], gap="large")
+    flowgraph_col, metrics_and_nodes_info_col = home_tab_cols
+
+    with flowgraph_col:
+        streamlit.markdown("\n")
```

Comment:
why?

Reply:
This is akin to the magic numbers. I thought that it looked better this way. I will add a comment to make that clearer.